### PR TITLE
Fixing preservation of unsignedness for read_imageui functions

### PR DESCRIPTION
**Pull request by neildhickey**
Test for this was added by svenvh. I can add to this pull request though if desired.

### CODE REVIEW

**Comment by AlexeySotkin** [comment]
@neildhickey, could you fix clang-format and clang-tidy issues.

File: lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -800,16 +800,27 @@ bool SPIRVToLLVM::postProcessOCLBuiltinWithArrayArguments(
   return true;
 }
 
-static char getTypeSuffix(Type *T) {
-  char Suffix;
+static std::string getTypeSuffix(Type *T, SPIRVType *SpirT) {
```
Comment (AlexeySotkin) [inline lib/SPIRV/SPIRVReader.cpp]: Can we please do not mix SPIR and SPIR-V. It might be confusing.

File: lib/SPIRV/SPIRVReader.cpp [hunk 2]
```diff
@@ -800,16 +800,27 @@ bool SPIRVToLLVM::postProcessOCLBuiltinWithArrayArguments(
...
   if (ST->isHalfTy())
-    Suffix = 'h';
+    Suffix = "h";
   else if (ST->isFloatTy())
-    Suffix = 'f';
-  else
-    Suffix = 'i';
+    Suffix = "f";
+  else {
+    SPIRVTypeInt *SIT = nullptr;
```
Comment (AlexeySotkin) [inline lib/SPIRV/SPIRVReader.cpp]: What does "SIT" means? Is it "SPIRV Integer Type"?

File: lib/SPIRV/SPIRVReader.cpp (hunk 1 above)
Comment (neildhickey) [reply lib/SPIRV/SPIRVReader.cpp]: Agreed sorry, will fix

File: lib/SPIRV/SPIRVReader.cpp (hunk 2 above)
Comment (neildhickey) [reply lib/SPIRV/SPIRVReader.cpp]: Yes, I'll rename it to STI, though for some reason that seemed to be a wrong thing to say :)

**Pull request merged**